fabric-convention-tags-v2/src/testmod/java/net/fabricmc/fabric/test/tag/convention/v2/TagUtilTest.java

```diff
@@ -16,26 +16,145 @@
 
 package net.fabricmc.fabric.test.tag.convention.v2;
 
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import org.objectweb.asm.Opcodes;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import net.minecraft.block.Blocks;
 import net.minecraft.enchantment.Enchantments;
 import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.tag.TagKey;
 import net.minecraft.world.biome.BiomeKeys;
 
 import net.fabricmc.api.ModInitializer;
 import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
 import net.fabricmc.fabric.api.tag.convention.v2.ConventionalBiomeTags;
 import net.fabricmc.fabric.api.tag.convention.v2.ConventionalBlockTags;
 import net.fabricmc.fabric.api.tag.convention.v2.ConventionalEnchantmentTags;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalEntityTypeTags;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalFluidTags;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalItemTags;
+import net.fabricmc.fabric.api.tag.convention.v2.ConventionalStructureTags;
 import net.fabricmc.fabric.api.tag.convention.v2.TagUtil;
 
 public class TagUtilTest implements ModInitializer {
 	private static final Logger LOGGER = LoggerFactory.getLogger(TagUtilTest.class);
 
+	private static final Class<?>[] TAG_CLASSES = {
+			ConventionalBiomeTags.class,
+			ConventionalBlockTags.class,
+			ConventionalEnchantmentTags.class,
+			ConventionalEntityTypeTags.class,
+			ConventionalFluidTags.class,
+			ConventionalItemTags.class,
+			ConventionalStructureTags.class,
+	};
+
+	// Think twice before choosing an inconsistent name
+	private static final Set<TagKey<?>> IGNORED_TAGS = Set.of(
+			ConventionalBiomeTags.IS_CONIFEROUS_TREE,
+			ConventionalBiomeTags.IS_DECIDUOUS_TREE,
+			ConventionalBiomeTags.IS_JUNGLE_TREE,
+			ConventionalBiomeTags.IS_SAVANNA_TREE,
+			ConventionalBiomeTags.IS_VEGETATION_DENSE,
+			ConventionalBiomeTags.IS_VEGETATION_DENSE_OVERWORLD,
+			ConventionalBiomeTags.IS_VEGETATION_SPARSE,
+			ConventionalBiomeTags.IS_VEGETATION_SPARSE_OVERWORLD,
+
+			ConventionalBlockTags.RED_SANDSTONE_BLOCKS,
+			ConventionalBlockTags.RED_SANDSTONE_SLABS,
+			ConventionalBlockTags.RED_SANDSTONE_STAIRS,
+			ConventionalBlockTags.UNCOLORED_SANDSTONE_BLOCKS,
+			ConventionalBlockTags.UNCOLORED_SANDSTONE_SLABS,
+			ConventionalBlockTags.UNCOLORED_SANDSTONE_STAIRS,
+
+			ConventionalItemTags.RED_SANDSTONE_BLOCKS,
+			ConventionalItemTags.RED_SANDSTONE_SLABS,
+			ConventionalItemTags.RED_SANDSTONE_STAIRS,
+			ConventionalItemTags.UNCOLORED_SANDSTONE_BLOCKS,
+			ConventionalItemTags.UNCOLORED_SANDSTONE_SLABS,
+			ConventionalItemTags.UNCOLORED_SANDSTONE_STAIRS
+	);
+
+	private static boolean isValidTagField(Field field) {
+		if ((field.getModifiers() & Opcodes.ACC_STATIC) == 0) {
```

Comment:
Nit: [`Modifier.STATIC`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/reflect/Modifier.html#STATIC) is better here as it's part of the same reflection API instead of ASM

Reply:
You can even use `Modifier.isStatic()` 👍